docs/architecture/adr-112-proposer-based-timestamps.md

```diff
@@ -125,56 +127,61 @@ type Vote struct {
 
 The consensus parameters will be updated to include this `Synchrony` field as follows:
 
 ```diff
 type ConsensusParams struct {
-	Block     BlockParams     `json:"block"`
-	Evidence  EvidenceParams  `json:"evidence"`
-	Validator ValidatorParams `json:"validator"`
-	Version   VersionParams   `json:"version"`
-++	Synchrony SynchronyParams `json:"synchrony"`
+        Block     BlockParams     `json:"block"`
+        Evidence  EvidenceParams  `json:"evidence"`
+        Validator ValidatorParams `json:"validator"`
+        Version   VersionParams   `json:"version"`
+        ABCI      ABCIParams      `json:"abci"`
+++      Synchrony SynchronyParams `json:"synchrony"`
 }
 ```
 
 ```go
 type SynchronyParams struct {
-	MessageDelay time.Duration `json:"message_delay"`
-	Precision    time.Duration `json:"precision"`
+        Precision    time.Duration `json:"precision,string"`
+        MessageDelay time.Duration `json:"message_delay,string"`
 }
 ```
 
 ### Changes to the block proposal step
 
 #### Proposer selects block timestamp
 
-CometBFT currently uses the `BFTTime` algorithm to produce the block's `Header.Timestamp`.
-The [proposal logic](https://github.com/cometbft/cometbft/blob/68ca65f5d79905abd55ea999536b1a3685f9f19d/internal/state/state.go#L269) sets the weighted median of the times in the `LastCommit.CommitSigs` as the proposed block's `Header.Timestamp`.
+CometBFT currently uses the `BFT Time` algorithm to produce the block's `Header.Timestamp`.
+The [block production logic](https://github.com/cometbft/cometbft/blob/1f430f51f0e390cd7c789ba9b1e9b35846e34642/internal/state/state.go#L248)
+sets the weighted median of the times in the `LastCommit.CommitSigs` as the proposed block's `Header.Timestamp`.
 
-In proposer-based timestamps, the proposer will still set a timestamp into the `Header.Timestamp`.
-The timestamp the proposer sets into the `Header` will change depending on if the block has previously received a [polka](https://github.com/cometbft/cometbft/blob/053651160f496bb44b107a434e3e6482530bb287/docs/introduction/what-is-cometbft.md#consensus-overview) or not.
+In PBTS, the proposer will still set a timestamp into the `Header.Timestamp`.
+The timestamp the proposer sets into the `Header` will change depending on if the block has previously received a Polka, i.e., `2/3+` prevotes in a previous round.
 
-#### Proposal of a block that has not previously received a polka
+#### Proposal of a block that has not previously received a Polka
 
 If a proposer is proposing a new block then it will set the Unix time currently known to the proposer into the `Header.Timestamp` field.
 The proposer will also set this same timestamp into the `Timestamp` field of the `Proposal` message that it issues.
```

Comment:
Do we check at any point that the timestamp set to the proposal matches the proposed block's timestamp?

Reply:
Yes, at the prevote step.